framework/audio/iaudioplayer.h

```diff
@@ -16,53 +16,25 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 //=============================================================================
+
 #ifndef MU_AUDIO_IAUDIOPLAYER_H
 #define MU_AUDIO_IAUDIOPLAYER_H
 
-#include "modularity/imoduleexport.h"
-
-#include "async/channel.h"
-#include "audiotypes.h"
-#include "midi/miditypes.h"
-
-//! NOTE This is the main public playback control interface for consumers,
-//! so namespace is just mu::audio
+#include <memory>
+#include "iplayer.h"
+#include "iaudiosource.h"
+#include "iaudiostream.h"
 
 namespace mu {
 namespace audio {
-class IAudioPlayer : MODULE_EXPORT_INTERFACE
+class IAudioPlayer : public IPlayer, public IAudioSource
```

Comment:
What is a goal to divide this interface into IAudioPlayer and IPlayer?

Reply:
@RomanPudashkin IPlayer inherited in IAudioPlayer and in IMIDIPlayer

Reply:
What is the difference between IAudioPlayer and IMIDIPlayer? Seems that we can make one interface (IAudioPlayer as was before) and inherit from it MIDIPlayer and other players

Reply:
IAudioPlayer loads IAudioStream (for example .ogg file)
IMIDIPlayer loads MidiStream (midi data) and has additional control methods for playing midi data.

Inheritance changed here https://github.com/musescore/MuseScore/pull/7272